File: leetcode/python/14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        min_length = min([len(s) for s in strs])
        common_prefix = ""
        for i in range(min_length):
            for s in strs:
                if s[i] != strs[0][i]:
                    return common_prefix
            common_prefix = common_prefix + strs[0][i]
        
        return common_prefix


solution = Solution()

input = ["flower","flow","flight"]
input = ["dog","racecar","car"]
print(f"in: {input}\nout: {solution.longestCommonPrefix(input)}")

File: leetcode/cpp/14.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 1)
        {
            return strs[0];
        }
        char ch;
        int idx = 0;
        while(1)
        {
            ch = strs[0][idx];
            for(int i = 1; i < strs.size(); i++)
            {
                if(strs[i][idx] != ch)
                {
                    return strs[i].substr(0, idx);                    
                }
            }
            idx += 1;
        }

    }
};

int main()
{
    Solution solution;
    vector<string> strs={"",""};
    
    string output = solution.longestCommonPrefix(strs);

    cout<<output<<endl;
    return 0;
}